ktor-io/jvm/src/io/ktor/utils/io/JvmSerializable.jvm.kt

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright 2014-2024 JetBrains s.r.o and contributors. Use of this source code is governed by the Apache 2.0 license.
+ */
+
+package io.ktor.utils.io
+
+import java.io.*
+
+public actual typealias JvmSerializable = Serializable
+
+@Suppress("UNCHECKED_CAST")
+public actual fun <T : Any> JvmSerializerReplacement(serializer: JvmSerializer<T>, value: T): Any =
+    DefaultJvmSerializerReplacement(serializer, value)
+
+@PublishedApi // IMPORTANT: changing the class name would result in serialization incompatibility
+internal class DefaultJvmSerializerReplacement<T : Any>(
+    private var serializer: JvmSerializer<T>?,
+    private var value: T?
+) : Externalizable {
```

Comment:
Should such an API better be integrated as part of the Kotlin stdlib itself or at least in a separate library for reuse in other projects? I'll add the same code to my open-source lib, too (https://github.com/ensody/ReactiveState-Kotlin).

Reply:
Regarding `JvmSreializable` interface, you can vote on this issue:
- [KT-48243](https://youtrack.jetbrains.com/issue/KT-48243) Publish `kotlin.io.Serializable`

I haven't found any issue proposing multiplatform `Externalizable` implementation, so feel free to create one!